openjdk8/src/main/docker/shutdown/shutdown-wrapper.bash

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+source /setup-env.d/05-utils.bash
+
+function java_shutdown_hook {
+  JAVA_PID=`ps -C java -o pid --no-headers`
+  PUSER=`ps -o ruser --no-headers -p $JAVA_PID`
+
+  # send thread dump to a file
+  if is_true "${SHUTDOWN_LOGGING_THREAD_DUMP}"; then
+    export THREAD_DUMP_FILE=${THREAD_DUMP_FILE:-/app.shutdown.threads}
+    # run jcmd under JVM process owner and skip first 3 lines of output
+    su $PUSER -c "jcmd $JAVA_PID Thread.print" | tail -n +3 &> $THREAD_DUMP_FILE
```

Comment:
Would the current user always have enough privilege to login as the owner of java ?

For example if the user is changed to `jetty` by the end user (Dockerfile) and the java executable is own by `root`.